controllers/provider-alicloud/pkg/controller/infrastructure/terraform.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package infrastructure
+
+import (
+	"github.com/gardener/gardener-extensions/controllers/provider-alicloud/pkg/apis/alicloud/v1alpha1"
+	"github.com/gardener/gardener-extensions/pkg/controller"
+	extensionsterraformer "github.com/gardener/gardener-extensions/pkg/gardener/terraformer"
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ComputeCreateVPCInitializerValues computes the InitializerValues to create a new VPC.
```

Comment:
IMO all functions in this file should be unexported. As far as I understand, they are currently only exported in order to be tested. However, you are already testing them as part of the actuator implementation, so testing them independently is redundant.

Alternatively, if you would rather prefer keeping them and the tests for them, you could implement them as the methods of a type (or types), e.g. `valuesProvider` for the fist 3, and inject this type via an interface into the actuator.

In summary, I believe with dependency injection you should aim at the following 3 cases:

* Unexported functions that are used to provide internal structure for your package. They are not tested independently and could be refactored into a different set of functions at any moment without any tests being broken.
* Explicitly declared dependencies via interfaces that are always mocked away in tests.
* Inexplicit dependencies to utility functions / types that are intended for reuse by different packages. These are typically "simple" enough to justify direct usage instead of injection via an interface.

As a result, there should not be exported utility functions / types in the package that are used solely by this package.